startups/lib/main_9.dart

```diff
@@ -0,0 +1,220 @@
+import 'package:english_words/english_words.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (BuildContext context) => MyAppState(),
```

Comment:
Having a single ChangeNotifierProvider at the top of the app is, in my opinion, the most straightforward way of simple app state management. And we do need app state management because we have several widgets.

Yes, we could have ValueNotifiers passed via constructors but that makes the code clunky. Yes, we could also have just setState everywhere, but the code gets pretty ugly even for a relatively straightforward app as this one.

Reply:
I find threading setState through widgets a lot harder to reason about compared to a single ChangeNotifier object holding the core app state.